src/app/onion/learning-object-builder/pages/materials-page/materials-page.component.ts

```diff
@@ -28,6 +29,16 @@ export class MaterialsPageComponent implements OnInit, OnDestroy {
     ).subscribe(val => {
       this.store.serviceInteraction$.next(val);
     });
+
+    // listen for outcome events and update component stores
+    this.store.learningObjectEvent
+      .pipe(takeUntil(this.destroyed$))
+      .subscribe((payload: LearningObject) => {
+        if (payload) {
```

Comment:
This is a nitpick, but when using RxJS I think it's better we leverage the API and reactive style as much as possible. The pipe could be rewritten as
```
.pipe(
  filter(learningObject => learningObject !== undefined),
  takeUntil(this.destroyed$)
)
```

This would let us avoid the if statement we have here all together, and keep our procedural logic minimized to what really matters - assigning the local value to the incoming data.

Reply:
This is actually code that was in place before this issue was opened, originally it was written like this because there was an else condition, but with that being removed it can be rewritten.